packages/dashboard-frontend/src/services/workspace-client/devworkspace/devWorkspaceClient.ts

```diff
@@ -53,6 +50,46 @@ export type Subscriber = {
   }
 };
 
+/**
+ * Context provided to all editors when they need to process devfile
+ */
+export interface IDevWorkspaceEditorProcessingContext {
+
+  devfile: devfileApi.Devfile;
+
+  devWorkspace: devfileApi.DevWorkspace;
+
+  workspaceId: string;
+
+  devWorkspaceTemplates: devfileApi.DevWorkspaceTemplateLike[];
+
+  editorsDevfile: devfileApi.Devfile[];
+
+  pluginRegistryUrl?: string;
+
+  axios: AxiosInstance;
+
+  optionalFilesContent: { [fileName: string]: string };
```

Comment:
I would replace it with a func that can provide content on-demand by doing requests to Che Server.

Reply:
I think it would be better to have this loading/patching stuff occurring on the server side as well.

Reply:
dashboard backend.
For example from client side, there is no need to download theia library or to load the default editor file
The content of the default editor should be loaded by the server side of dashboard (and the theia library as well)